scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/edf/EDFPolicyExtended.java

```diff
@@ -0,0 +1,157 @@
+ * If job is not started, we sort them by `effectiveDeadline - (now() + expectedTime`.
+ * If deadline is absolute then effectiveDeadline equals to deadline.
+ * If deadline is relative then effectiveDeadline is calculated as (now() + deadline)
+ *
+ * Among jobs without deadline (and the same priority), we sort them by submission date.
+ *
+ * @author The ProActive Team
+ * @since ProActive Scheduling 8.3
+ */
+public class EDFPolicyExtended extends ExtendedSchedulerPolicy {
+
+    private static final Date MAXIMUM_DATE = new Date(Long.MAX_VALUE);
+
+    @Override
+    public LinkedList<EligibleTaskDescriptor> getOrderedTasks(List<JobDescriptor> jobs) {
+        final Date now = new Date();
+
+        final Comparator<JobDescriptor> jobDescriptorComparator = (job1, job2) -> {
+
+            final InternalJob internalJob1 = ((JobDescriptorImpl) job1).getInternal();
+            final InternalJob internalJob2 = ((JobDescriptorImpl) job2).getInternal();
+            JobPriority job1Priority = internalJob1.getPriority();
+            JobPriority job2Priority = internalJob2.getPriority();
+            if (!job1Priority.equals(job2Priority)) {
+                return job2Priority.compareTo(job1Priority);
+            } else {
+                if (internalJob1.getJobDeadline().isPresent() & !internalJob2.getJobDeadline().isPresent()) {
+                    return -1;
+                } else if (!internalJob1.getJobDeadline().isPresent() & internalJob2.getJobDeadline().isPresent()) {
+                    return 1;
+                } else if (!internalJob1.getJobDeadline().isPresent() & !internalJob2.getJobDeadline().isPresent()) {
+                    return Long.compare(internalJob1.getJobInfo().getSubmittedTime(),
+                                        internalJob2.getJobInfo().getSubmittedTime());
+                } else { // if(internalJob1.getJobDeadline().isPresent() & internalJob2.getJobDeadline().isPresent()) {
+                    if (internalJob1.getJobInfo().getStartTime() >= 0 &&
+                        internalJob2.getJobInfo().getStartTime() >= 0) {
+                        // both started
+                        return Long.compare(internalJob1.getJobInfo().getStartTime(),
+                                            internalJob2.getJobInfo().getStartTime());
+                    } else if (internalJob1.getJobInfo().getStartTime() >= 0 &&
+                               internalJob2.getJobInfo().getStartTime() < 0) {
+                        // priority to already started - internalJob1
+                        return -1;
+                    } else if (internalJob1.getJobInfo().getStartTime() < 0 &&
+                               internalJob2.getJobInfo().getStartTime() >= 0) {
+                        // priority to already started - internalJob2
+                        return 1;
+                    } else {//if(internalJob1.getJobInfo().getStartTime() < 0 && internalJob2.getJobInfo().getStartTime() < 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)